pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -495,16 +495,24 @@ func (lb *ListenerBuilder) buildSidecarOutboundListeners(node *model.Proxy,
 							// Make sure each endpoint address is a valid address
 							// as service entries could have NONE resolution with label selectors for workload
 							// entries (which could technically have hostnames).
-							if !netutil.IsValidIPAddress(instance.Address) {
+							isValidInstance := true
+							for _, addr := range instance.Addresses {
+								if !netutil.IsValidIPAddress(addr) {
+									isValidInstance = false
+									break
+								}
+							}
+							if !isValidInstance {
 								continue
 							}
 							// Skip build outbound listener to the node itself,
 							// as when app access itself by pod ip will not flow through this listener.
 							// Simultaneously, it will be duplicate with inbound listener.
-							if instance.Address == node.IPAddresses[0] {
+							nodeKey := strings.Join(node.IPAddresses, ",")
```

Comment:
This is not right... node.IPAddress can have many many addresses. On my machine I have 25 addresses there.

These are not comparable.

Reply:
I checked in real cluster, this behavior is broken

Reply:
Hi @howardjohn, thanks for this validation, I change this code as below:

```
var shouldContinue bool = false
// should continue if current IstioEndpoint instance has the same ip with any of node IPaddresses
for _, naddr := range node.IPAddresses {
    if instance.Key() == naddr {
        shouldContinue = true
    }
}
if shouldContinue {
    continue
}
```
it should be skip if current IstioEndpoint instance's Key() is contained in node IPAddresses. Does it make sense?

Reply:
Hmm... I am not sure its valid to compare to all node.IPAddresses. Some of these are private IPs, no?

Also, why do we only check the Key() (i.e. first IP) and not all of them?

Reply:
Hi @howardjohn, okay, we can keep the same as we change the IstioEndpoint Key(), only check the first IP, it should be more stable.